jdbc/src/main/java/org/apache/metamodel/jdbc/dialects/ImpalaQueryRewriter.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.jdbc.dialects;
+
+import org.apache.metamodel.jdbc.JdbcDataContext;
+
+/**
+ * Query rewriter for Apache Impala
+ **/
+public class ImpalaQueryRewriter extends Hive2QueryRewriter{
```

Comment:
Is Impala's SQL dialect so closely related to Hive's SQL dialect that it should inherit it directly? Or is it that they should have a common super class? I think that when we choose inheritance it's always important to ask/verify that "X is a type of Y", not just that they share attributes or things. Like "A cat is an animal" because it _IS_ an animal, but say the class Dog defined "four legs", I would still advise against doing a "Cat extends Dog" relationship.

Reply:
Because Impala uses the same metadata store as Hive to record information about table structure and properties，so Impala provides a high degree of compatibility with the Hive Query Language (HiveQL) ;
[(http://impala.apache.org/docs/build/html/topics/impala_langref.html)]

Reply:
OK that makes sense to me then. But can you then please add that to the class's javadoc comments so that this relationship is clear to anyone reading the code.

Reply:
Ok,I will do that.